interface/controller/pkmn_controller.go

```diff
@@ -0,0 +1,49 @@
+package controller
```

Comment:
This comment applies for all the errors in general.

While it's ok to return when an error happens, what it's usually done is to log the error you got and then respond to the user with an error that would be more appropriate.

Logging allows us to debug from our side what's going on since logs are usually aggregated into a logging service, and it also allows us to not return an error so explicit to the user that could potentially give out details on how our APIs work. So after logging whatever error we got, we want to return to the user a custom error.

For example, if you didn't find the resource the user was requesting it would make sense to return a 404 - not found with a message that goes like `pokemon not found`, or when something internal failed, like failing to read the csv file, or failing to connect with a DB, then we would just send the user a 500 status code with a message like `something went wrong!`, while logging as much info as we can to help us debug if necessary.

So TL;DR, log your errors and return something more user friendly as a response

Reply:
Added a general purpose logger to address this.